### PR TITLE
Translatable page

### LINKED ISSUE #331: Meta title/description for CMS page not translatable by default
I just found out that a CMS page's Meta title and Description is not translatable by default. 
schermafbeelding 2018-01-11 om 10 30 23

Only the slug is translatable. 
Am I missing something here?
Or might it be a good thing to make those fields also translatable by default?

### PR DESCRIPTION
**Pull request by mjauvin**
- Add CMS Page translation (title, description, meta-title and meta-description)
- Fix a typo in "hasTranslatableAttributes" method name

Resolves #331

### CODE REVIEW

**Comment by munxar** [comment]
@mjauvin 👍 thank you!

**Comment by munxar** [comment]
@LukeTowers @mjauvin tested this feature. looks good.

File: classes/TranslatableBehavior.php [hunk 1]
```diff
@@ -297,7 +297,7 @@ public function lang($context = null)
      * Checks if this model has transatable attributes.
      * @return true
      */
-    public function hasTransatableAttributes()
+    public function hasTranslatableAttributes()
```
Comment (LukeTowers) [inline classes/TranslatableBehavior.php]: You need to update the call to the method here as well: https://github.com/rainlab/translate-plugin/blob/e603c487fa0c1c4a9ca31cd6c1a9305ea2364389/classes/EventRegistry.php#L84 do a search of the code to see if it was misspelled anywhere else too.

Comment (LukeTowers) [reply classes/TranslatableBehavior.php]: Never mind, I see you fixed it in https://github.com/rainlab/translate-plugin/pull/363/files#diff-252810a1b0bce82b1a1ef724cee3c8deR96

**Comment by LukeTowers** [comment]
@munxar could you find all the issues for this that have been posted in the octobercms/october, rainlab/pages-plugin, and rainlab/translate-plugin and link them here so that I can close all of them when I merge this?

**Comment by mjauvin** [comment]
@LukeTowers Hmmm, rainlab.pages uses '@RainLab.Translate.Behaviors.TranslatablePageUrl' and I removed it... I'll add it back. Would be nice to completely remove it, though, what's the best course of action?

**Comment by munxar** [comment]
@LukeTowers I think these are the related ones:
- https://github.com/rainlab/translate-plugin/issues/331
- https://github.com/rainlab/translate-plugin/issues/205
- https://github.com/rainlab/translate-plugin/issues/138

@mjauvin Imho the RainLab.Pages plugin should not depend on the translate plugin, but the other way round.
for all other controls the instantiation call comes from the translate plugin, but here (https://github.com/rainlab/pages-plugin/blob/master/classes/Content.php) the pages plugin does it.

shouldn't we move this into the pages plugin? @LukeTowers @mjauvin Wha do you think?

Btw. you could mark the TranslatableUrl as depreciated just in case someone else is using it.

File: updates/version.yaml [hunk 2]
```diff
@@ -49,3 +49,4 @@
 1.3.7: Fix config reference in LocaleMiddleware
 1.3.8: Keep query string when switching locales
 1.4.0: Add importer and exporter for messages
+1.4.1: Add TranslatablePage behavior (replaces TranslatablePageUrl)
```
Comment (munxar) [inline updates/version.yaml]: remove the (replace TranslatablePageUrl)

**Comment by mjauvin** [comment]
@LukeTowers anything I can do to help close this PR?

File: updates/version.yaml [hunk 3]
```diff
@@ -49,3 +49,6 @@
 1.3.7: Fix config reference in LocaleMiddleware
 1.3.8: Keep query string when switching locales
 1.4.0: Add importer and exporter for messages
+1.4.1:
```
Comment (LukeTowers) [inline updates/version.yaml]: Is this form of release notes actually supported by October? Typically multi-line versions are to provide one comment about the version and any DB migration scripts to run.

File: classes/EventRegistry.php [hunk 4]
```diff
@@ -29,25 +29,36 @@ public function registerFormFieldReplacements($widget)
...
         }
 
-        if (
-            $model instanceof Page &&
-            isset($widget->fields['settings[url]'])
-        ) {
-            $widget->fields['settings[url]']['type'] = 'mltext';
+        if ($model instanceof Page)
+        {
+            if (isset($widget->fields['settings[title]']))
```
Comment (LukeTowers) [inline classes/EventRegistry.php]: Is there a cleaner way to do this? Why can't we use https://github.com/rainlab/translate-plugin/blob/master/classes/EventRegistry.php#L107?

File: updates/version.yaml (hunk 3 above)
Comment (mjauvin) [reply updates/version.yaml]: You may be right, I'll change this.

**Comment by mjauvin** [comment]
Closed and moved to #367

**Pull request closed without merging**